content/blogposts/2024/top-5-tips-for-using-cody-with-react.md

```diff
@@ -0,0 +1,306 @@
+      </div>
+      <div className="input_container">
+        <SearchInput
+          placeHolder="Enter Keywords"
+          value={keywords}
+          handleChange={handleInputChange}
+        />
+        <SubmitButton
+          buttonText="Generate Tweet"
+          handleClick={handleInputSubmit}
+        />
+      </div>
+      <Footer />
+    </div>
+  );
+};
+
+export default Home;
+```
+
+### 2. Input Clear Prompts
+
+The second tip is that whenever you use Cody, constantly input clear and concise prompts. Let's look into some example prompts that we will use with our app to make this clear. 
+
+I input Cody this prompt: `I want to use the keywords from the input to generate the Tweet content. Write me the code for it.`
+
+<table>
+  <tr>
+    <td><img src="https://github.com/deepak2431/cody-blogs/raw/react-cody-tip/blogs/assets/prompt_5_1.jpg" width="280" height="320" /></td>
+    <td><img src="https://github.com/deepak2431/cody-blogs/raw/react-cody-tip/blogs/assets/prompt_5_2.jpg" width="280" height="320" /></td>
+  </tr>
+ </table>
+
+You can see that it was not able to generate any code as it is asking for more clear prompt details so can provide helpful code which can be used. What if we clarify in the prompt that it should be for the React app?
+
+The updated Cody prompt: `I want to use the keywords from the input to generate the Tweet content using the openai npm package in a React app. Write me the code for it.`
+
+<table>
+  <tr>
+    <td><img src="https://github.com/deepak2431/cody-blogs/raw/react-cody-tip/blogs/assets/prompt_6_1.jpg" width="280" height="320" /></td>
+    <td><img src="https://github.com/deepak2431/cody-blogs/raw/react-cody-tip/blogs/assets/prompt_6_2.jpg" width="280" height="320" /></td>
+    <td><img src="https://github.com/deepak2431/cody-blogs/raw/react-cody-tip/blogs/assets/prompt_6_3.jpg" width="280" height="320" /></td>
+  </tr>
+ </table>
+
+A small changes in prompt made huge difference in the result. The results it gave is mostly perfect which can be used. Awesome job by Cody here! 
+
+This is the `generateTweet` function for our app:
+
+```typescript
+  const generateTweet = async() => {
+
+    const response = await openai.chat.completions.create({
+      messages: [{ role: 'user', content: `Generate a viral tweet using the keywords: ${keywords}` }],
+      model: 'gpt-3.5-turbo',
+    })
+
+    const tweet = response.choices[0].message.content;
+
+    console.log(tweet)
+  }
+```
+
+This is why it is always good to keep in mind that you add as much clarity as you can in the prompts to get a better response. 
+
+So now, we are done with 80% of our app, and we need to render the generated results using the Twitter Card UI component.
+
+### 3. Use commands effectively
+
+The commands is one of the favourite features of Cody for many users because of it being so developer-friendly to use. You can use commands to ask a question, fix a bug, make a refactor or edit any portion of your code.
```

Comment:
```suggestion
[Commands](https://sourcegraph.com/docs/cody/capabilities/commands) is one of my favorite features of Cody because they are so developer-friendly. You can use commands to ask a question, fix a bug, refactor your code, or edit any portion of your code.
```